view-generator-framework/src/main/java/org/hypertrace/core/viewgenerator/service/MultiViewGeneratorLauncher.java

```diff
@@ -66,7 +66,7 @@ public List<String> getInputTopics(Map<String, Object> properties) {
       Config viewGenConfig = viewGenConfigs.get(viewGen);
       inputTopics.addAll(viewGenConfig.getStringList(INPUT_TOPICS_CONFIG_KEY));
     }
-    return inputTopics.stream().collect(Collectors.toList());
+    return new ArrayList<>(inputTopics);
```

Comment:
nit - why this change? List.copyOf (unless we're intentionally making this mutable)

That said, we should make this deterministic and avoid the hash set here.

(same comment below)

Reply:
> why this change?

Intellij suggestion

> avoid the hash set here

set needed as input topic across all viewgens is same and we need to add only one source per input topic.

Reply:
> set needed as input topic across all viewgens is same and we need to add only one source per input topic.

So we have a uniqueness requirement, not a set it sounds like. Using a hash set means that different runs with the same input will return their outputs in a different order. Even if that's not impactful today, it's a bad practice and makes writing tests annoying. Suggest any of
- use a list, than run distinct on it
- use an insertion-ordered set (e.g. guava's or linkedhashset I believe)
- use streams e.g.
```java
return getViewGenName(properties)
  .stream()
  .map(viewGenConfigs::get)
  .map(config -> config.getStringList(INPUT_TOPICS_CONFIG_KEY))
  .flatMap(Collection::stream)
  .distinct()
  .collect(Collectors.toUnmodifiableList());
```

Reply:
Why not a TreeSet? 
Set for me ensures uniqueness and TreeSet ensures ordering (for tests thingy you mentioned).

Looks simpler and much readable than the streams snippet in the suggestion.

Reply:
> Looks simpler and much readable than the streams snippet in the suggestion.

matter of opinion! but yes, any ordered set will do for repeatability, doesn't need to be insertion ordered - I was just trying to match the perceived intent.